x-pack/plugins/cloud_security_posture/server/plugin.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  PluginInitializerContext,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  Logger,
+} from '../../../../src/core/server';
+import { createFindingsIndexTemplate } from './index_template/create_index_template';
+import type {
+  CspServerPluginSetup,
+  CspServerPluginStart,
+  CspServerPluginSetupDeps,
+  CspServerPluginStartDeps,
+} from './types';
+import { defineRoutes } from './routes';
+import { initUiSettings } from './ui_settings';
+
+export class CspPlugin
+  implements
+    Plugin<
+      CspServerPluginSetup,
+      CspServerPluginStart,
+      CspServerPluginSetupDeps,
+      CspServerPluginStartDeps
+    >
+{
+  private readonly logger: Logger;
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+  }
+
+  public setup(
+    core: CoreSetup<CspServerPluginStartDeps, CspServerPluginStart>,
+    plugins: CspServerPluginSetupDeps
+  ): CspServerPluginSetup {
+    this.logger.debug('csp: Setup');
+    const router = core.http.createRouter();
+
+    // Register server side APIs
+    defineRoutes(router, this.logger);
+    initUiSettings(core.uiSettings);
+
+    return {};
+  }
+
+  public start(core: CoreStart, plugins: CspServerPluginStartDeps): CspServerPluginStart {
+    this.logger.debug('csp: Started');
```

Comment:
nit: You don't need to mention the name of your plugin in any of your server-side logs, core will automatically add it to the logging context for you.

Actually, you don't really need to indicate that the plugin has started, either... core logs that by default too :)